python/lsst/meas/extensions/ngmix/processCoaddsNGMix.py

```diff
@@ -109,53 +112,58 @@ def cdict(self):
 
-        So far all children are u sing this exact code so leaving
+        So far all children are using this exact code so leaving
         it here for now. If we specialize a lot, might make a
         processor its own object
 
         Parameters
         ----------
-        images : `dict` of `lsst.afw.image.ExposureF`
-            Coadd images and associated metadata, keyed by filter name.
-        ref : `lsst.afw.table.SourceCatalog`
+        catexps : `Dict [CatalogExposure]`
+            List of coadd images and associated catalogs/metadata.
+        cat_ref : `lsst.afw.table.SourceCatalog`
             A catalog with one record for each object, containing "best"
             measurements across all bands.
-        replacers : `dict` of `lsst.meas.base.NoiseReplacer`, optional
-            A dictionary of `~lsst.meas.base.NoiseReplacer` objects that can
-            be used to insert and remove deblended pixels for each object.
-            When not `None`, all detected pixels in ``images`` will have
-            *already* been replaced with noise, and this *must* be used
-            to restore objects one at a time.
-        imageId : `int`
-            Unique ID for this unit of data.  Should be used (possibly
-            indirectly) to seed random numbers.
-        butler : `lsst.daf.persistence.Butler`
-            Butler to output incremental results to if enabled.
-        kwargs_butler : `dict`
-            Additional keyword arguments to pass to `butler`.put() for
-            incremental output; default {}.
+
         Returns
         -------
         results : `lsst.pipe.base.Struct`
             Struct with (at least) an `output` attribute that is a catalog
             to be written as ``self.config.output``.
         """
-
         tm0 = time.time()
         nproc = 0
 
-        self.set_rng(imageId)
+        self.set_rng(self.config.seed)
 
         config = self.cdict
         self.log.info(pprint.pformat(config))
 
+        n_images = len(catexps)
+        images = [None]*n_images
+        replacers = [None]*n_images
+        replace = config['useDeblends']
+        for idx, catexp in enumerate(catexps):
+            images[idx] = catexp.exposure
+            if replace:
+                catalog = catexp.catalog
+                footprints = {rec.getId(): (rec.getParent(), rec.getFootprint()) for rec in catalog}
+                # This is hideous, but I'm not sure how else to do it
+                try:
+                    exposureId = catalog.getMetadata().getScalar(SingleFrameMeasurementTask.NOISE_EXPOSURE_ID)
+                except Exception:
+                    exposureId = hash(catalog)
+                replacer = NoiseReplacer(self.config.deblendReplacer, exposure=catexp.exposure,
+                                         footprints=footprints, exposureId=exposureId)
+                catexp.metadata['noiseReplacer'] = replacer
+            replacers[idx] = catexp.metadata.get('noiseReplacer')
```

Comment:
Will that key exist if replace is not True?

Reply:
No, hence the `get` (which returns `None` by default if the key doesn't exist).